### DISCUSSION TITLE
Allow adding http-only forges

### DISCUSSION THREAD

**Discussion by peacememories** [magit/forge#667] [Support] [answered]
At work we have a private Gitlab server, which does not support HTTPS. The reasoning is somewhere between "It's only accesible on the internal network" and "We don't want to deal with private CAs". In any case I have no control over it.

When trying to add this server to forge I put `http://` in front of the API and web hosts, which then resulted in forge creating the URL `https://http://<our url>/api/v4`. Trying again, I just used the host name, but this, predictably, tried to access `https://<our url>/api/v4`.

It would be really helpful to be able to specify the protocol, or for forge to recognize a leading protocol string and not try to add its own.

**Comment by tarsius** [comment, marked as answer]
Since https://github.com/magit/forge/commit/9f152cfa803e6b77121a2779c930b0e64a472463 you can just add an entry to `ghub-insecure-hosts`, to achieve that.

**Reply by peacememories** [reply]
oh, thanks :) I'll try that tomorrow

**Reply by peacememories** [reply]
Hmm, it seems I'm doing something wrong. I set the insecure host using `(setq ghub-insecure-hosts '("<our host>"))` and this indeed makes forge announce that it would connect to `http://<our host>/api/v4` instead of `https://...`.

When I go to add the repository, however, I get the following error which still reads like it's trying to connect to port 443:

```
open-gnutls-stream: make client process failed: Connection refused, :name, <our host>, :buffer, #<killed buffer>, :host, <our host>, :service, 443, :nowait, nil, :noquery, nil, :tls-parameters, nil, :coding, nil
```

**Reply by tarsius** [reply]
> Hmm, it seems I'm doing something wrong.

Maybe but it's also possible there's a bug. I have no way of testing this, as I do not have access to a private instance.

> I get the following error

Could I please get a backtrace with that? If you don't mind me seeing the information you retracted above, but don't want to publish it for everyone to see, you can mail it to me. (The rest of the conversation I would still want to have here.) If the backtrace seems unreadable, listing a lot of "compiled functions", then visit the relevant libraries (at least `forge-gitlab.el`), and do `M-x eval-buffer` there, and then repeat the request.

> "<our host>"

It might help if you posted the unredacted values of `ghub-insecure-hosts` and `forge-alist`, or if you could use something closer to the actual value, along the lines of "example.com".
